internal/handler/handler_test.go

```diff
@@ -486,3 +488,146 @@ func TestTransactionManagementService_GetTransactions(t *testing.T) {
 		})
 	}
 }
+func TestTransactionManagementService_DownloadTransaction(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	tests := []struct {
+		name  string
+		model model.GetTransaction
+		setup func() (*transactionManagementService, *http.Request)
+		want  func(recorder httptest.ResponseRecorder)
+	}{
+		{
+			name: "Success",
+			setup: func() (*transactionManagementService, *http.Request) {
+				mockLogic := mock.NewMockTransactionManagementServiceLogicIer(mockCtrl)
+				mockLogic.EXPECT().DownloadTransaction(gomock.Any(), "456").Times(1).Return(&respModel.Response{
+					Status:  http.StatusOK,
+					Message: codes.GetErr(codes.Success),
+					Data:    []byte("PDF"),
+				})
+				svc := &transactionManagementService{
+					logic: mockLogic,
+				}
+
+				r := httptest.NewRequest("GET", "/transactions/download/:123", nil)
+
+				ctx := session.SetSession(r.Context(), model.SessionStruct{UserId: "1234", Cookie: "456"})
+				return svc, r.WithContext(ctx)
+			},
+			want: func(rec httptest.ResponseRecorder) {
```

Comment:
Also validate the Headers that are being set in the code.

Reply:
done